app/questionnaire/rules.py

```diff
@@ -185,18 +197,26 @@ def _get_answers_on_routing_path(schema, routing_path, answer_store, repeat_rule
             if answer_id in answer_ids_on_path:
                 repeat_indexes.append(answer_id)
 
-    answers = list(answer_store.filter(answer_ids=repeat_indexes))
+    answers = answer_store.filter(answer_ids=repeat_indexes)
 
     # Exclude answers that are not on the routing path from repeat function
-    answers_to_remove = []
-    for answer in answers:
-        if not _is_answer_on_path(schema, answer, routing_path):
-            answers_to_remove.append(answer)
+    answers_on_path = _get_answers_on_path(answers, schema, routing_path)
+
+    return answers_on_path
+
+
+def _get_answers_on_path(answers, schema, routing_path) -> AnswerStore:
```

Comment:
Could this just return a list of the answers? then we wouldnt need to `.copy()` the answer store?

Reply:
Currently, yes, though that actually raises another issue. I hadn't changed `get_answer_store_value` to use the `answers_on_path` instead of `filtered`. It would have worked most of the time.

Reply:
So, after that change, we need to call filter on the result of `_get_answers_on_path`